controllers/deactivation/deactivation_controller_test.go

```diff
@@ -86,76 +84,78 @@ func TestReconcile(t *testing.T) {
+			// when
+			timeSinceProvisioned := time.Since(murProvisionedTime.Time)
 
-				// then
-				require.NoError(t, err)
+			// when
+			_, err := r.Reconcile(context.TODO(), req)
 
-				// Reload the usersignup
-				reloaded := &toolchainv1alpha1.UserSignup{}
-				err = cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, reloaded)
-				require.NoError(t, err)
+			// then
+			require.NoError(t, err)
 
-				// Confirm the deactivating state has been unset in the UserSignup
-				require.False(t, states.Deactivating(reloaded))
+			// Reload the usersignup
+			err = cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar)
+			require.NoError(t, err)
 
-				t.Run("reconcile a third time to confirm the scheduled deactivation time is correctly set", func(t *testing.T) {
-					// given
-					r, req, cl = prepareReconcile(t, mur.Name, userTier30, mur, reloaded, config)
+			// Confirm the deactivating state has been unset in the UserSignup
+			require.False(t, states.Deactivating(userSignupFoobar))
 
-					// when
-					// Reconcile again
-					res, err := r.Reconcile(context.TODO(), req)
-					require.NoError(t, err)
+			t.Run("reconcile again to confirm the scheduled deactivation time is correctly set", func(t *testing.T) {
+				// when
+				res, err := r.Reconcile(context.TODO(), req)
+				require.NoError(t, err)
 
-					// then
-					expectedTime := (time.Duration((expectedDeactivationTimeoutDeactivate30Tier-preDeactivationNotificationDays)*24) * time.Hour) - timeSinceProvisioned
-					actualTime := res.RequeueAfter
-					diff := expectedTime - actualTime
-					require.Truef(t, diff > 0 && diff < 2*time.Second, "expectedTime: '%v' is not within 2 seconds of actualTime: '%v' diff: '%v'", expectedTime, actualTime, diff)
-					assertThatUserSignupStateIsDeactivated(t, cl, username, false)
+				// then
+				expectedTime := (time.Duration((expectedDeactivationTimeoutDeactivate30Tier-preDeactivationNotificationDays)*24) * time.Hour) - timeSinceProvisioned
+				actualTime := res.RequeueAfter
+				diff := expectedTime - actualTime
+				require.Truef(t, diff > 0 && diff < 2*time.Second, "expectedTime: '%v' is not within 2 seconds of actualTime: '%v' diff: '%v'", expectedTime, actualTime, diff)
+				assertThatUserSignupStateIsDeactivated(t, cl, username, false)
 
-					// confirm that the scheduled deactivation time is set
-					require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
-					require.NotNil(t, userSignupFoobar.Status.ScheduledDeactivationTimestamp)
+				// confirm that the scheduled deactivation time is set
+				require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
+				require.NotNil(t, userSignupFoobar.Status.ScheduledDeactivationTimestamp)
```

Comment:
if you update the function `assertThatUserSignupStateIsDeactivated` so it returns the retrieved Usersignup:
```go
func assertThatUserSignupStateIsDeactivated(t *testing.T, cl *commontest.FakeClient, name string, expected bool) *toolchainv1alpha1.UserSignup {
	userSignup := &toolchainv1alpha1.UserSignup{}
	err := cl.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: operatorNamespace}, userSignup)
	require.NoError(t, err)
	require.Equal(t, expected, states.Deactivated(userSignup))
	return userSignup
}
```
then you could simplify this code:
```
userSignupFoobar = assertThatUserSignupStateIsDeactivated(t, cl, username, false)
```

Reply:
good suggestion, changed.